mycpp/gc_alloc.h

```diff
@@ -22,6 +22,40 @@ extern MarkSweepHeap gHeap;
 #define VALIDATE_ROOTS 0
 
 // mycpp generates code that keeps track of the root set
+class StackRoot {
+ public:
+  StackRoot(void* root) {
+#if VALIDATE_ROOTS
+    RawObject* obj = *(reinterpret_cast<RawObject**>(root));
+    if (obj) {
+      RawObject* header = ObjHeader::FromObject(obj);
```

Comment:
Also, by inspection, I wonder if we had a compile error here

I think this should be `ObjHeader* header`, not `RawObject*`

So it's masked by the `#ifdef` ?   Another reason to extract it, and put only the call site in the `#ifdef`, not the body

Reply:
Yup. There was :) fixed in the new commit